app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt

```diff
@@ -27,114 +26,95 @@ import androidx.compose.material3.Surface
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
 import com.infomaniak.swisstransfer.R
 import com.infomaniak.swisstransfer.ui.components.*
-import com.infomaniak.swisstransfer.ui.images.AppImages.AppIcons
 import com.infomaniak.swisstransfer.ui.images.AppImages.AppIllus
-import com.infomaniak.swisstransfer.ui.images.icons.DocumentOnDocument
-import com.infomaniak.swisstransfer.ui.images.icons.PersonBadgeShare
 import com.infomaniak.swisstransfer.ui.images.illus.beers.Beers
 import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.components.TransferType
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.upload.components.ShareAndCopyButtons
 import com.infomaniak.swisstransfer.ui.theme.Dimens
 import com.infomaniak.swisstransfer.ui.theme.Margin
 import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
 import com.infomaniak.swisstransfer.ui.utils.PreviewAllWindows
-import com.infomaniak.swisstransfer.ui.utils.copyText
-import com.infomaniak.swisstransfer.ui.utils.shareText
-import kotlinx.coroutines.launch
 
 @Composable
 fun UploadSuccessQrScreen(transferType: TransferType, transferUrl: String, closeActivity: () -> Unit) {
-
-    val context = LocalContext.current
     val snackbarHostState = remember { SnackbarHostState() }
 
     BottomStickyButtonScaffold(
         snackbarHostState = snackbarHostState,
         topBar = { BrandTopAppBar() },
-        topButton = {
-            LargeButton(
-                modifier = it,
-                style = ButtonType.PRIMARY,
-                titleRes = R.string.buttonShare,
-                imageVector = AppIcons.PersonBadgeShare,
-                onClick = { context.shareText(transferUrl) },
-            )
-        },
         bottomButton = {
             LargeButton(
                 modifier = it,
-                style = ButtonType.SECONDARY,
+                style = ButtonType.PRIMARY,
                 titleRes = R.string.buttonFinished,
                 onClick = closeActivity,
             )
         },
-        content = { Content(context, snackbarHostState, transferType, transferUrl) },
-    )
+    ) {
+        Column {
+            ScreenContent(transferType, transferUrl)
+
+            ShareAndCopyButtons(
+                modifier = Modifier.padding(bottom = Margin.Medium, top = Margin.Mini),
+                transferLink = transferUrl,
+                snackbarHostState = snackbarHostState,
+            )
+        }
+    }
 }
 
 @Composable
-private fun Content(context: Context, snackbarHostState: SnackbarHostState, transferType: TransferType, transferUrl: String) {
-
-    val scope = rememberCoroutineScope()
-
+private fun ColumnScope.ScreenContent(
```

Comment:
This screen content composable seems overkill, also I don't really like this because the ShareAndCopyButton are screen content too, but aren't inside the composable named this way